feat: Localize dates, bytes, and numbers

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix/issues/2143

## Changes
- Defaults to user's browser language preference when displaying dates, bytes, and numbers (with caveats.) This fixes an issue where numbers were always formatted in English.
- Shows user's browser language preferences in language dropdown
- Fixes timezone not displayed in archived item detail start and finish times
- Standardizes number formatting
- Sets timezone that unit tests run in

## Manual testing

Tested with Firefox and Chrome

1. Set language preference in browser of choice
2. Log in
3. Verify dates, bytes, and numbers are displayed correctly in chosen language
4. Go to Account Settings
5. Change language preference to a language that displays numbers differently
6. Go back to org dashboard. Verify date/byte/numbers are displayed correctly

## Screenshots

| Page | Image/video |
| ---- | ----------- |
| Account Settings | <img width="1024" alt="Screenshot 2024-11-12 at 3 22 51 PM" src="https://github.com/user-attachments/assets/6fcabef9-c278-43d6-ab3f-d3d18e2a4a58"> |
| Collections | (Browser language set to Korean, no app language set)<br/><img width="1292" alt="Screenshot 2024-11-12 at 3 22 07 PM" src="https://github.com/user-attachments/assets/8a264713-66ec-4a40-b56b-953f93f74796"> |
| Archived Item Detail | <img width="200" alt="Screenshot 2024-11-12 at 3 44 13 PM" src="https://github.com/user-attachments/assets/c56862d1-877b-4b88-8321-8f70c586b26e"> |


<!-- ## Follow-ups -->

SuaYoo committed:
prevent directly setting locale

SuaYoo committed:
localize numbers

SuaYoo committed:
localize bytes

SuaYoo committed:
remove unused user language

SuaYoo committed:
fix timezone

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
switch to html lang

SuaYoo committed:
Apply `localize:extract` changes

SuaYoo committed:
standardize with shoelace localize

emma-sg reviewed (approved):
Looks great! Tested locally, all instances that I could find look good in both languages.

emma-sg commented on frontend/src/features/archived-items/crawl-pending-exclusions.ts:
```diff
@@ -72,7 +72,7 @@ export class CrawlPendingExclusions extends LiteElement {
         ${this.total
           ? this.total > 1
-            ? msg(str`+${this.total.toLocaleString()} URLs`)
+            ? msg(str`+${this.total} URLs`)
```
This should be rendered formatted as well, I think — lemme start a branch for updates to this

emma-sg replied:
#2160

emma-sg committed:
Frontend localize dates updates (#2160)

Suggestions for #2146

Ended up being just one conversion, but it's a pain to do in the
suggestion editor thingy.

---------

Co-authored-by: emma-sg <emma-sg@users.noreply.github.com>

SuaYoo committed:
add test

SuaYoo committed:
fix linting issue

SuaYoo committed:
fix tests

emma-sg committed:
clean up type overrides

SuaYoo committed:
update ts comments

emma-sg committed:
add descriptions to `@ts-expect-error`s & fix unused import

Shrinks99 reviewed (commented):
Tested locally, by in large looks like a great change and a big step forward for localization support! 🎉 

Was only able to find one issue!

Using the pagination on the crawl workflows list seems to result in an error.  None of the other pages experience this problem.  This does not happen in main.  I made sure to run `yarn` to build new packages.

<img width="1188" alt="Screenshot 2024-11-19 at 5 44 09 PM" src="https://github.com/user-attachments/assets/b1281ad8-de74-446b-9ce0-dc4e4fb2229b">

SuaYoo committed:
fix name rendering

SuaYoo commented:
> Using the pagination on the crawl workflows list seems to result in an error. None of the other pages experience this problem. This does not happen in main. I made sure to run `yarn` to build new packages.

Nice catch, fixed!

SuaYoo committed:
upgrade localize tools

SuaYoo committed:
Apply `localize:extract` changes
